react-components/src/components/RevealContainer/RevealResourcesContext.tsx

```diff
@@ -0,0 +1,38 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import { type CogniteModel } from '@cognite/reveal';
+import { type ReactElement, createContext, useContext, useState, type ReactNode } from 'react';
+
+type RevealResourcesContextType = {
+  reveal3DResources: CogniteModel[];
+  update3DResources: (newResources: CogniteModel[]) => void;
+};
+
+export const RevealResourcesContext = createContext<RevealResourcesContextType | null>(null);
+
+export const useRevealResources = (): RevealResourcesContextType => {
+  const element = useContext(RevealResourcesContext);
+  if (element === null) {
+    throw new Error('useRevealResources must be used within a RevealResourcesContextProvider');
+  }
+  return element;
+};
+
+export const RevealResourcesContextProvider = ({
+  children
+}: {
+  children: ReactNode;
+}): ReactElement => {
+  const [reveal3DResources, setRevealResources] = useState<CogniteModel[]>([]);
```

Comment:
Do we really need a reference to the whole model? I guess we just want to track the addition/deletion of models and we don't use actual models. We can get the model by id from `viewer.models`. So maybe to make it slightly easier we can only store modelId and revisionId inside this context.